### PR TITLE
Adding Locale object as parameter to DeliveryParameterBuilder

### PR DESCRIPTION
**Pull request by cunhazera**
Solve this: https://github.com/Kentico/delivery-sdk-java/issues/28

### CODE REVIEW

**Comment by cunhazera** [comment]
@aweigold I getting an error with the build on travis. Do you know what it is?

File: src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java [hunk 1]
```diff
@@ -179,6 +180,13 @@ public DeliveryParameterBuilder language(String language) {
         return this;
     }
 
+    public DeliveryParameterBuilder language(Locale language) {
+        if (language != null) {
+            nameValuePairs.add(new BasicNameValuePair(LANGUAGE, language.toString()));
```
Comment (aweigold) [inline src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: This might not work.  Although you can manually make a locale in the "<language>-<variant>" syntax, you are actually only setting the language doing so.  Constructing language with a variant with a Java locale is done via one of the following examples:

```
Locale.UK
new Locale("en", "GB")
```

The toString() method on Locale will return a locale formatted as "<language>_<variant>", which doesn't match the spec from the Kentico API.  Unless that format is supported, this method will need to do the proper string formatting.

File: src/test/java/com/kenticocloud/delivery/DeliveryParameterBuilderTest.java [hunk 2]
```diff
@@ -272,8 +272,17 @@ public void testLanguage() {
     }
 
     @Test
+    public void testLocaleLanguage() {
+        List<NameValuePair> params = DeliveryParameterBuilder.params().language(new Locale("pt-br")).build();
```
Comment (aweigold) [inline src/test/java/com/kenticocloud/delivery/DeliveryParameterBuilderTest.java]: This test is only setting the language "pt-br" instead of the language "pt" with variant "br".

File: src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java (hunk 1 above)
Comment (cunhazera) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: @aweigold so `new Locale("en", "GB")` must return 'en-GB' ?

Comment (aweigold) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: That can work.  You will also want to unit test an edge case where no variant is set, so if just new Locale("en") is passed, you are not sending in "en-".

**Comment by aweigold** [comment]
The PR build issue is fixed with #31

File: src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java (hunk 1 above)
Comment (cunhazera) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: @aweigold if the client passes only `new Locale("en")` is must return only "en"?

Comment (aweigold) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: Correct

Comment (cunhazera) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: So it's working correctly

Comment (aweigold) [reply src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: @cunhazera "So it's working correctly" as in the Kentico API will accept the underscores?  Or it's working correctly if you just replace the "_" for "-" ?  If the latter, that replacement will work, I was just advising that it would make sense to do a unit test for that case as well.

File: src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java [hunk 3]
```diff
@@ -182,7 +182,7 @@ public DeliveryParameterBuilder language(String language) {
 
     public DeliveryParameterBuilder language(Locale language) {
         if (language != null) {
-            nameValuePairs.add(new BasicNameValuePair(LANGUAGE, language.toString()));
+            nameValuePairs.add(new BasicNameValuePair(LANGUAGE, language.toString().replace('_', '-')));
```
Comment (aweigold) [inline src/main/java/com/kenticocloud/delivery/DeliveryParameterBuilder.java]: 👍

**Comment by aweigold** [comment]
Looks like I spoke too soon on the PR fix... this is good to merge in.  Thanks!

**Comment by cunhazera** [comment]
@aweigold thank you, man.

**Comment by petrsvihlik** [comment]
thank you @cunhazera and @aweigold !

**Comment by petrsvihlik** [comment]
Hi @cunhazera, 
we would like to send you some Kentico Cloud stickers as a thanks for taking part in Hacktoberfest. If you want them, please send us your postal address at developerscommunity@kentico.com.

**Comment by cunhazera** [comment]
Hello @petrsvihlik I just sent an email with my address. Thank you so much!
